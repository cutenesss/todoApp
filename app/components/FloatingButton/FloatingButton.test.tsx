import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react-native';
import { FloatingButton } from '.';

describe("<FloatingButton />", () => {
    const mockOnPress = jest.fn();
    test('renders correctly', () => {
        const tree = renderer.create(<FloatingButton onPress={mockOnPress} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('onPress works correctly', () => {
        render(<FloatingButton onPress={mockOnPress} />);
        fireEvent.press(screen.getByText("+"));
        expect(mockOnPress).toBeCalledTimes(1);
    });
});