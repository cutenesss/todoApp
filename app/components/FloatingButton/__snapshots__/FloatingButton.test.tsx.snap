// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FloatingButton /> renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "blue",
      "borderRadius": 40,
      "bottom": 30,
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "position": "absolute",
      "right": 30,
      "width": 40,
      "zIndex": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 25,
          "fontWeight": "bold",
        },
        Object {
          "color": "#fff",
        },
      ]
    }
  >
    +
  </Text>
</View>
`;
